https://github.com/Exiv2/exiv2/commit/82adcb1f86def0555b25713a8b84dba84944b275 (1)
https://github.com/Exiv2/exiv2/commit/7d8cd00778c3eee9f32af04b009d8a5651eb53e7 (2)

This is basically just commit (1). Though it wasn't directly applicable so I just merged it with (2).
Easier and it makes the file same as upstream.

diff -u b/unitTests/test_futils.cpp b/unitTests/test_futils.cpp
--- b/unitTests/test_futils.cpp
+++ b/unitTests/test_futils.cpp
@@ -42,21 +42,8 @@
     std::string tmpFile("tmp.dat");
     std::ofstream auxFile(tmpFile.c_str());
     auxFile.close();
-#if   defined(_WIN32) || defined(__CYGWIN__) || defined(__MINGW__) || defined(__MSYS__)
-    const char * expectedString = "No error (errno = 0)";
-#elif defined(__APPLE__)
-    const char * expectedString = "Undefined error: 0 (errno = 0)";
-#elif defined(__sun__)
-    const char * expectedString = "Error 0 (errno = 0)";
-#elif defined(__FreeBSD__)
-    const char * expectedString = "No error: 0 (errno = 0)";
-#elif defined(__NetBSD__)
-    const char * expectedString = "Undefined error: 0 (errno = 0)";
-#else
-    const char * expectedString = "Success (errno = 0)";
-#endif
     std::remove(tmpFile.c_str());
-    ASSERT_STREQ(expectedString, strError().c_str());
+    ASSERT_TRUE(strError().find("(errno = 0)") != std::string::npos);
 }
 
 TEST(strError, returnNoSuchFileOrDirectoryWhenTryingToOpenNonExistingFile)
@@ -68,22 +55,7 @@
 TEST(strError, doNotRecognizeUnknownError)
 {
     errno = 9999;
-#if   defined(__MINGW__) || defined(__MSYS__) || defined(__CYGWIN__)
-    const char * expectedString = "Unknown error 9999 (errno = 9999)";
-#elif defined(_WIN32)
-    const char * expectedString = "Unknown error (errno = 9999)";
-#elif defined(__APPLE__)
-    const char * expectedString = "Unknown error: 9999 (errno = 9999)";
-#elif defined(__sun__)
-    const char * expectedString = "Unknown error (errno = 9999)";
-#elif defined(__FreeBSD__)
-    const char * expectedString = "Unknown error: 9999 (errno = 9999)";
-#elif defined(__NetBSD__)
-    const char * expectedString = "Unknown error: 9999 (errno = 9999)";
-#else
-    const char * expectedString = "Unknown error 9999 (errno = 9999)";
-#endif
-    ASSERT_STREQ(expectedString, strError().c_str());
+    ASSERT_TRUE(strError().find("(errno = 9999)") != std::string::npos);
 }
 
 TEST(getEnv, getsDefaultValueWhenExpectedEnvVariableDoesNotExist)
